lib/insights_cloud/async/insights_scheduled_sync.rb

```diff
@@ -0,0 +1,25 @@
+require 'rest-client'
+
+module InsightsCloud
+  module Async
+    class InsightsScheduledSync < ::ApplicationJob
+      def perform
+        unless Setting[:allow_auto_insights_sync]
+          logger.debug(
+            'The scheduled process is disabled due to the "allow_auto_insights_sync"
+            setting being set to false.'
+          )
+          return
+        end
+
+        InsightsFullSync.perform_later()
+      ensure
+        self.class.set(:wait => 24.hours).perform_later
```

Comment:
So if the setting is disabled, this job will keep on firing daily even though it won't really do anything?

Reply:
Yes, the job is responsible to reschedule itself. This way I don't need to monitor settings and enable the schedule once the setting changes.